test/oci-tests/tests/common/mod.rs

```diff
@@ -0,0 +1,173 @@
+            Error::Others(format!(
+                "failed to read requested Wasm module ({}): {}. Perhaps you need to run 'make test/wasm-modules' first.",
+                name, e
+            ))
+        })
+}
+
+/*
+// TODO: Enable wasmtime test runner when OCI support is enabled
+fn run_wasmtime_test_with_spec(
+    dir: &TempDir,
+    spec: &Spec,
+    wasmbytes: &[u8],
+) -> Result<(u32, DateTime<Utc>), Error> {
+    create_dir(dir.path().join("rootfs"))?;
+
+    let wasm_path = dir.path().join("rootfs").join(WASM_FILENAME);
+    let mut f = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .mode(0o755)
+        .open(wasm_path)?;
+    f.write_all(wasmbytes)?;
+
+    let stdout = File::create(dir.path().join("stdout"))?;
+    drop(stdout);
+
+    spec.save(dir.path().join("config.json"))?;
+
+    let mut cfg = InstanceConfig::new(WasmtimeWasi::new_engine()?, "test_namespace".into());
+    let cfg = cfg
+        .set_bundle(dir.path().to_str().unwrap().to_string())
+        .set_stdout(dir.path().join("stdout").to_str().unwrap().to_string());
+
+    let wasi = Arc::new(WasmtimeWasi::new("test".to_string(), Some(cfg)));
+
+    wasi.start()?;
+
+    let (tx, rx) = channel();
+    let waiter = Wait::new(tx);
+    wasi.wait(&waiter).unwrap();
+
+    let res = match rx.recv_timeout(Duration::from_secs(60)) {
+        Ok(res) => Ok(res),
+        Err(e) => {
+            wasi.kill(SIGKILL as u32).unwrap();
+            return Err(Error::Others(format!(
+                "error waiting for module to finish: {0}",
+                e
+            )));
+        }
+    };
+
+    wasi.delete()?;
+    res
+}
+*/
+
+fn run_wasmedge_test_with_spec(
+    dir: &TempDir,
+    spec: &Spec,
+    wasmbytes: &[u8],
+) -> Result<(u32, DateTime<Utc>), Error> {
+    create_dir(dir.path().join("rootfs"))?;
+    let rootdir = dir.path().join("runwasi");
+    create_dir(&rootdir)?;
+    let opts = Options {
+        root: Some(rootdir),
+    };
+    let opts_file = OpenOptions::new()
+        .read(true)
+        .create(true)
+        .truncate(true)
+        .write(true)
+        .open(dir.path().join("options.json"))?;
+    write!(&opts_file, "{}", serde_json::to_string(&opts)?)?;
+
+    let wasm_path = dir.path().join("rootfs").join(WASM_FILENAME);
+    let mut f = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .mode(0o755)
+        .open(wasm_path)?;
+    f.write_all(wasmbytes)?;
+
+    let stdout = File::create(dir.path().join("stdout"))?;
+    drop(stdout);
+
+    spec.save(dir.path().join("config.json"))?;
+
+    let mut cfg = InstanceConfig::new(WasmEdgeWasi::new_engine()?, "test_namespace".into());
+    let cfg = cfg
+        .set_bundle(dir.path().to_str().unwrap().to_string())
+        .set_stdout(dir.path().join("stdout").to_str().unwrap().to_string());
+
+    let wasi = Arc::new(WasmEdgeWasi::new("test".to_string(), Some(cfg)));
```

Comment:
I believe this returns the `trait Instance`?  If so we could invert this.  Create the `config` and the `instance` outside the module and make this function generic `run_test_with_spec` and pass in the module.  

This would make the tests written below to not be specific calls to each runtime `let (output, retval) = common::run_wasmedge_test(&spec, &bytes)?;` making it more of a framework.